### PR TITLE
f/637 fix comparison screen

### LINKED ISSUE #637: new comparison design crashes with resolutions below 4k
**Please describe the bug:**  
the new comparison screen makes it impossible to read numbers, when more then 1 db is selected at resolutions below 4k.

**Please describe steps to reproduce:**  
1. comparison screen
2. See error

**What is the expected behavior? Please describe:**  
the text should be readable on all resolution >=1080p

**Screenshot(s):**  
image

**Affected Component(s):**  
Frontend

### PR DESCRIPTION
**Pull request by MaxSchneider1337**
# Resolves #637

**Does your pull request solve a problem? Please describe:**  
fixes the comparison screen design, because it only works on xl resolutions

**Affected Component(s):**  
Frontend

**Additional context:**  
image

### CODE REVIEW

**Comment by MaxSchneider1337** [comment]
without some of the whitespace
image
image

**Comment by Bouncner** [comment]
Shouldn't that safely fit into two rows?

**Comment by MaxSchneider1337** [comment]
> Shouldn't that safely fit into two rows?
 
2 rows does not work with my resolution, since Data Size and Memory Capacity Strings can get pretty long.

3 selected Databases would also destroy the formatting in that case (edited)

**Comment by Bensk1** [comment]
> 2 rows does not work with my resolution, since Data Size and Memory Capacity Strings can get pretty long.

We could display the data size in GB, that should save a bit of space. If necessary, we could also go for `# of CPUs`?

**Comment by MaxSchneider1337** [comment]
> Shouldn't that safely fit into two rows?

The comparison-screen only looks like this on resolutions below 1904px

**Comment by Bouncner** [comment]
So it looks like it on the most popular MacBook? ("13-inch MacBook Pro models introduced in late 2012 or later. Native resolution: 2560 x 1600 at 227 pixels per inch. Support for millions of colors.")

**Comment by MaxSchneider1337** [comment]
> So it looks like it on the most popular MacBook? ("13-inch MacBook Pro models introduced in late 2012 or later. Native resolution: 2560 x 1600 at 227 pixels per inch. Support for millions of colors.")

Yes and we are using every single color ;-)

I will try a different layout and let you now once it is ready

**Comment by MaxSchneider1337** [comment]
@Bouncner what do you think about this design: 
image
image

**Comment by MaxSchneider1337** [comment]
> @Bouncner what do you think about this design:
> image
> image

we could also

**Comment by Bouncner** [comment]
If it takes up that much space, we should redesign it or make is "minimizable" IMO. That's just too much space wasted for static information.

**Comment by Bouncner** [comment]
Also, can we group the metric correctly? Host name, host stats (i.e., capacity and cpus), footprint, and workers.

**Comment by MaxSchneider1337** [comment]
> Also, can we group the metric correctly? Host name, host stats (i.e., capacity and cpus), footprint, and workers.

yes of cause

File: hyrisecockpit/frontend/src/components/details/DatabaseSystemDetails.vue [hunk 1]
```diff
@@ -6,48 +6,84 @@
...
-        <v-row>
-          <v-col id="database-host" class="leftcol" cols="3">
-            <v-card-title class="ml-6 font-weight-regular white--text">
+      <v-card class="card" :color="database.color">
+        <v-row class="detailsRow">
+          <v-col
+            id="database-host"
+            :lg="databases.length > 1 ? 4 : 3"
+            xl="3"
+            v-bind:class="{
```
Comment (schTi) [inline hyrisecockpit/frontend/src/components/details/DatabaseSystemDetails.vue]: `v-bind` prefix can go here, `:` is basically the shortcut for this.

**Review by schTi — COMMENTED** [review]
Looks great. Is there anything missing you want to add?

**Comment by MaxSchneider1337** [comment]
> Looks great. Is there anything missing you want to add?

I don't really know... The fix is already done so this issue is ready. The best approach to me would be to write a new issue for improving the design, since this would be a pretty big task.

what do you think @Bouncner ?

**Comment by Bouncner** [comment]
Is it possible to scale the font for all characteristics (not the host name) with the available width. The additional lines really bother me. They are additional information and take up by far too much space at the beginning of the page and with their background they absolutely pull the user's attention.

**Comment by monasobh** [comment]
I also think we should make the font smaller. If all the metric numbers still don't fit into one row, I would suggest to leave the host name and - depeding on how many fit - one or two metric numbers in one row and add a button on the side of the page to be able to expand the db information boxes.

**Comment by MaxSchneider1337** [comment]
> Is it possible to scale the font for all characteristics (not the host name) with the available width. The additional lines really bother me. They are additional information and take up by far too much space at the beginning of the page and with their background they absolutely pull the user's attention.

scaling the text does not really work for me, because the text gets really small with 3 dbs.

**Comment by MaxSchneider1337** [comment]
> Is it possible to scale the font for all characteristics (not the host name) with the available width. The additional lines really bother me. They are additional information and take up by far too much space at the beginning of the page and with their background they absolutely pull the user's attention.

image
 this is how it looks with scaled text, really sorry for all the confusion this caused, but I just realized, that the MacBook scales the browsersize to 1440x900 px

**Comment by Bouncner** [comment]
Can we use "Workers" and "CPUs" ... I think the "number of" is useless.

I still think the fonts can be smaller though but that's up to you.

**Comment by MaxSchneider1337** [comment]
image

This is the current version on a small screen

**Comment by MaxSchneider1337** [comment]
> Can we use "Workers" and "CPUs" ... I think the "number of" is useless.
> 
> I still think the fonts can be smaller though but that's up to you.

already fixed this yesterday, forgot to push, sorry

**Comment by Bouncner** [comment]
I know

**Comment by Bouncner** [comment]
Whooopsie. Sorry.

**Comment by Bouncner** [comment]
I know, I am a pain in the ass. But can you try to increase the distinction between the host part on the left and the "remainder". I think we could use smaller fonts for the remainder and please also decrease the vertical space between the lines (Memory Capacity).

**Comment by MaxSchneider1337** [comment]
@Bouncner what do you think about this?

image

**Comment by Bouncner** [comment]
Maybe right align all the "remainders"? But that's up to you. It's just that I think the space smallest space between all the boxes is between the title and the remainder. There should be the biggest gap.

Apart from that: LGTM.

**Comment by MaxSchneider1337** [comment]
image
this is how the screen looks now so the different chart header elements do not overlap on smaller screens. All of the icons show the db name on hover.

**Comment by MaxSchneider1337** [comment]
> image
> this is how the screen looks now so the different chart header elements do not overlap on smaller screens

**Review by schTi — APPROVED** [review]
Looks good, feel free to merge if you are happy :fire:

**Pull request merged**